Samples/InpaintHTTP/WebHost.cs

```diff
@@ -0,0 +1,131 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using Nancy;
+using Nancy.Hosting.Self;
+using Nancy.ErrorHandling;
+using System.Linq;
+using Zavolokas.GdiExtensions;
+using Zavolokas.ImageProcessing.Inpainting;
+using Zavolokas.Structures;
+using System.Threading.Tasks;
+using Zavolokas.ImageProcessing.PatchMatch;
+using System.Drawing;
+using System.IO;
+using System.Threading;
+//using Zavolokas.Utils.Processes;
+
+//TODO: Cleanup the mess above & unused References
+
+namespace InpaintHTTP
+{
+    class WebHost
+    {
+        public WebHost()
+        {
+            NancyHost host = new NancyHost(new Uri("http://localhost:8069"));
+            host.Start();
+            Console.WriteLine("Initialized and started WebHost");
+        }
+    }
+
+    public class MainMod : NancyModule
+    {
+        public MainMod()
+        {
+            Post("/api/inpaint", async x =>
+            {
+                Console.WriteLine("[" + DateTime.Now.ToString("HH:mm:ss.fffff") + "] Incomming request from " + this.Request.UserHostAddress);
+                if (this.Request.Files.Count() > 1)
+                {
+                    Bitmap BitmapImg;
+                    Bitmap BitmapMask;
+
+                    byte[] ByteImg = new byte[this.Request.Files.First().Value.Length];
```

Comment:
This code `this.Request.Files.First()`  is used 3 times within two lines. It would be good idea to set the file to a separate variable and use it. Moreover it will improve performance a bit since there will be only one chain of calls to get the file instead of 3.

```suggestion
                    var imageFile = this.Request.Files.First();
                    byte[] ByteImg = new byte[imageFile.Value.Length];
```